handshake_client.go

```diff
@@ -411,6 +414,48 @@ func (hs *clientHandshakeState) processCertsFromServer(certificates [][]byte) er
 	return nil
 }
 
+// processDelegatedCredentialFromServer unmarshals the DelegatedCredential
+// offered by the server (if present) and validates it using the peer
+// certificate.
+func (hs *clientHandshakeState) processDelegatedCredentialFromServer(dc []byte) error {
+	c := hs.c
+
+	var cred *DelegatedCredential
+	var err error
+	if dc != nil {
+		if !hs.hello.delegatedCredential {
+			c.sendAlert(alertUnexpectedMessage)
+			return errors.New("tls: got delegated_credential extension without indication")
+		}
+
+		cred, err = unmarshalDelegatedCredential(dc)
+		if err != nil {
+			c.sendAlert(alertDecodeError)
+			return fmt.Errorf("tls: delegated_credential: %s", err)
+		}
+	}
+
+	if cred != nil && !c.config.InsecureSkipVerify {
+		var now time.Time
+		if hs.c.config.Time != nil {
+			now = hs.c.config.Time()
+		} else {
+			now = time.Now()
+		}
+
+		if v, err := cred.Validate(c.peerCertificates[0], hs.c.vers, now); err != nil {
+			c.sendAlert(alertIllegalParameter)
+			return fmt.Errorf("delegated_credential: %s", err)
+		} else if !v {
+			c.sendAlert(alertIllegalParameter)
```

Comment:
Maybe nit-picking, but still - is "alertIllegalParameter" correct here?
I mean - I'm aware of 7 days validity "rule" for DC. But what if it doesn't verify correctly? Shouldn't it be "bad_certificate" as for TLS 1.3?

Reply:
This is what the spec says to do. I can go either way. Do you think the spec should be changed?

Reply:
Hmmm... which place in the spec do you refer to exactly?

Reply:
Actually, this change *did* land: https://github.com/tlswg/tls-subcerts/commit/f249f76fd2c34ccb526fcdfc3d296da329c6d510